subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -448,10 +469,86 @@ static int open(const struct device *instance)
 
 }
 
+static int get_tx_buffer(const struct device *instance, void *token,
+			 void **data, uint32_t *len, k_timeout_t wait)
+{
+	struct ipc_rpmsg_ept *rpmsg_ept;
+	uint32_t max_size;
+	void *payload;
+	int bin_wait;
+
+	if (!data || !len) {
+		return -EINVAL;
+	}
+
+	/* OpenAMP only supports a binary wait / no-wait */
+	if (!K_TIMEOUT_EQ(wait, K_FOREVER) && !K_TIMEOUT_EQ(wait, K_NO_WAIT)) {
+		return -EINVAL;
+	}
+
+	bin_wait = K_TIMEOUT_EQ(wait, K_FOREVER);
+	rpmsg_ept = (struct ipc_rpmsg_ept *) token;
+
+	payload = rpmsg_get_tx_payload_buffer(&rpmsg_ept->ep, &max_size, bin_wait);
+	if (!payload) {
+		return -EIO;
+	}
+
+	/*
+	 * We still want to return the payload buffer in case of -ENOMEM
+	 * because the user would want to drop it or just use it anyway with
```

Comment:
`... want to drop...` but how? tx drop not supported.

Reply:
not supported yet. This will be added in OpenAMP soon-ish.